subiquity/ui/views/welcome.py

```diff
@@ -39,20 +39,14 @@ def __init__(self, model, controller):
             Text(""),
             Padding.center_79(Text("(More language choices will appear in time)"))]))
 
-    def _build_buttons(self):
-        self.buttons = [
-            Color.button(ok_btn(on_press=self.confirm)),
-        ]
-        return Pile(self.buttons)
-
     def _build_model_inputs(self):
         sl = []
-        for lang in self.model.get_languages():
-            sl.append(Color.menu_button(menu_btn(label=lang, on_press=self.confirm)))
+        for lang, code in self.model.get_languages():
+            sl.append(menu_btn(label=lang, on_press=self.confirm, user_arg=code))
```

Comment:
Yeap, i see this now. I have ended up doing something similar in my i18n branch. But also hook-up actual switch languages code. Will rebase my code on top of this. So this is fine to go in now.

Reply:
Oh heh, well either way...